cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/MaintenanceTimeWindowsTest.java

```diff
@@ -551,4 +556,68 @@ public boolean certRenewed() {
 
         this.reconciliationState.topicOperatorDeployment(dateSupplier).setHandler(handler);
     }
+
+    @Test
+    public void testZkRollingUpdateMaintenanceNotSatisfiedCertNotChangedSomethingElseIs(TestContext context) {
+
+        Async async = context.async();
+
+        doZkRollingUpdate(Collections.singletonList("* * 8-10 * * ?"),
+                () -> Date.from(LocalDateTime.of(2018, 11, 26, 11, 00, 0).atZone(ZoneId.of("GMT")).toInstant()),
+                r -> {
+                    String generation = getClusterCaGenerationPod(ZookeeperCluster.zookeeperPodName(NAME, 0));
+                    context.assertEquals("1", generation, "Pod had unexpected generation " + generation);
+                    async.complete();
+                }, false, true);
+
+        async.await();
+    }
+
+    @Test
+    public void testZkRollingUpdateMaintenanceNotSatisfiedCertChangedSomethingElseIs(TestContext context) {
+
+        Async async = context.async();
+
+        doZkRollingUpdate(Collections.singletonList("* * 8-10 * * ?"),
+                () -> Date.from(LocalDateTime.of(2018, 11, 26, 11, 00, 0).atZone(ZoneId.of("GMT")).toInstant()),
+                r -> {
+                    String generation = getClusterCaGenerationPod(ZookeeperCluster.zookeeperPodName(NAME, 0));
+                    context.assertEquals("1", generation, "Pod had unexpected generation " + generation);
+                    async.complete();
+                }, true, true);
+
+        async.await();
+    }
+
+    @Test
+    public void testZkRollingUpdateMaintenanceNotSatisfiedCertChangedSomethingElseIsNot(TestContext context) {
+
+        Async async = context.async();
+
+        doZkRollingUpdate(Collections.singletonList("* * 8-10 * * ?"),
+                () -> Date.from(LocalDateTime.of(2018, 11, 26, 11, 00, 0).atZone(ZoneId.of("GMT")).toInstant()),
+                r -> {
+                    String generation = getClusterCaGenerationPod(ZookeeperCluster.zookeeperPodName(NAME, 0));
+                    context.assertEquals("0", generation, "Pod had unexpected generation " + generation);
+                    async.complete();
+                }, true, false);
+
+        async.await();
+    }
+
+    @Test
+    public void testZkRollingUpdateMaintenanceSatisfiedCertChangedSomethingElseIsNot(TestContext context) {
+
+        Async async = context.async();
+
+        doZkRollingUpdate(Collections.singletonList("* * 8-10 * * ?"),
+                () -> Date.from(LocalDateTime.of(2018, 11, 26, 9, 00, 0).atZone(ZoneId.of("GMT")).toInstant()),
+                r -> {
+                    String generation = getClusterCaGenerationPod(ZookeeperCluster.zookeeperPodName(NAME, 0));
+                    context.assertEquals("1", generation, "Pod had unexpected generation " + generation);
+                    async.complete();
+                }, true, false);
+
+        async.await();
+    }
```

Comment:
@ppatierno I added tests to try combinations as the tables at PR description show. I did it just for the Zk because all other resources follow the same path (`isPodToRestart`). Are you ok with that?

Reply:
@stanlyDoge sorry to be picky, I would do for Kafka as well :-)